[SDK-3849] Remove claimCheck from withAuthenticationRequired (#456)
auth0/auth0-react: MIGRATION_GUIDE.md

```

As you can see, `scope` becomes a merged value of the previous `defaultScope` and `scope`.

## Removal of `claimCheck` on `withAuthenticationRequired`

> In v1 of Auth0-React `withAuthenticationRequired` Higher Order Component supported a `claimCheck` property that would check the claims in a JWT against the provided set and redirect the user back to the Auth0 login page if they did not have the sufficient claims. Given that it is highly unlikely for a user role to change by logging in and that it would most likely lead to users being stuck in infinite login loops, we have chosen to remove this functionality from Auth0-React and instead provide guidance on how to achieve this so that developers can have greater control over the behavior of their application.

Reviewer: ```suggestion
In v1 of Auth0-React the `withAuthenticationRequired` Higher Order Component supported a `claimCheck` property that would check the ID Token's claims and redirect the user back to the Auth0 login page if the check failed. Given that it is unlikely for most user claims to change by logging in again, it would most likely lead to users being stuck in infinite login loops. Therefore, we have chosen to remove this functionality from Auth0-React and instead provide guidance on how to achieve this so that developers can have greater control over the behavior of their application.
```
